pandas/_testing.py

```diff
@@ -1866,6 +1873,18 @@ def assert_copy(iter1, iter2, **eql_kwargs):
         assert elem1 is not elem2, msg
 
 
+def is_ExtensionArrayDtype_and_needs_i8_conversion(left_dtype, right_dtype):
```

Comment:
can you rename to 

is_extension_array_dtype_and_needs_i8_conversion (e.g. all lowercase)

ideally type left_dtype, right_dtype (I think we type these in other places)

Reply:
and type the return type as bool